include/grpc++/impl/codegen/call.h

```diff
@@ -560,12 +562,21 @@ class CallOpClientRecvStatus {
 
   void FinishOp(bool* status) {
     if (recv_status_ == nullptr) return;
-    metadata_map_->FillMap();
+    trailing_metadata_map_->FillMap();
+    // In the case where we receive a Trailers-Only response, the
+    // metadata will be returned as initial metadata instead of trailing
+    // metadata, so we need to check both.
     grpc::string binary_error_details;
-    auto iter = metadata_map_->map()->find(kBinaryErrorDetailsKey);
-    if (iter != metadata_map_->map()->end()) {
+    auto iter = trailing_metadata_map_->map()->find(kBinaryErrorDetailsKey);
+    if (iter != trailing_metadata_map_->map()->end()) {
```

Comment:
Now that you are here. Could you erase the pair from the trailing/initial metadata map? I think this is an oversight from the previous implementation and now that it can appear in either trailing or initial metadata would make it more confusing. Thanks.